src/sql/parts/notebook/notebookActions.ts

```diff
@@ -461,40 +462,54 @@ export class AttachToDropdown extends SelectBox {
 	 **/
 	public async openConnectionDialog(useProfile: boolean = false): Promise<void> {
 		try {
-			await this._connectionDialogService.openDialogAndWait(this._connectionManagementService, { connectionType: 1, providers: this.model.getApplicableConnectionProviderIds(this.model.clientSession.kernel.name) }, useProfile ? this.model.connectionProfile : undefined).then(connection => {
-				let attachToConnections = this.values;
-				if (!connection) {
-					this.loadAttachToDropdown(this.model, this.getKernelDisplayName());
-					this.doChangeContext(undefined, true);
-					return;
-				}
-				let connectionUri = this._connectionManagementService.getConnectionUri(connection);
-				let connectionProfile = new ConnectionProfile(this._capabilitiesService, connection);
-				let connectedServer = connectionProfile.title? connectionProfile.title : connectionProfile.serverName;
-				//Check to see if the same server is already there in dropdown. We only have server names in dropdown
-				if (attachToConnections.some(val => val === connectedServer)) {
-					this.loadAttachToDropdown(this.model, this.getKernelDisplayName());
-					this.doChangeContext();
-					return;
-				}
-				else {
-					attachToConnections.unshift(connectedServer);
-				}
-				//To ignore n/a after we have at least one valid connection
-				attachToConnections = attachToConnections.filter(val => val !== msgSelectConnection);
-
-				let index = attachToConnections.findIndex((connection => connection === connectedServer));
-				this.setOptions([]);
-				this.setOptions(attachToConnections);
-				if (!index || index < 0 || index >= attachToConnections.length) {
-					index = 0;
-				}
-				this.select(index);
+			let connectionUri = generateUri(this.model.connectionProfile, 'notebook') + '|' + generateUuid();
+			await this._connectionDialogService.openDialogAndWait(this._connectionManagementService,
+				{
+					connectionType: 1,
+					providers: this.model.getApplicableConnectionProviderIds(this.model.clientSession.kernel.name),
+					input: <IConnectableInput>{
+						uri: connectionUri,
+						onConnectStart: undefined,
+						onConnectSuccess: undefined,
+						onConnectReject: undefined,
+						onDisconnect: undefined,
+						onConnectCanceled: undefined
+					},
+				},
+				useProfile ? this.model.connectionProfile : undefined).then(connection => {
+					this.model.addAttachToConnectionsToBeDisposed(connectionUri);
+					let attachToConnections = this.values;
+					if (!connection) {
```

Comment:
This means that the user didn't successfully make a connection right? Do you want to be calling addAttachToConnectionsToBeDisposed in that case?